### PR TITLE
Fix outline focus at Safari

### PR DESCRIPTION
**Pull request by mrtuvn**
https://github.com/argyleink/open-props/issues/212

### CODE REVIEW

File: src/extra/normalize.css [hunk 1]
```diff
@@ -37,7 +37,11 @@
   outline-offset: 5px;
 }
 
-:where(body) { 
+:where(input[type='text']:not([disabled]):focus) {
+  outline: none;
+}
```
Comment (argyleink) [inline src/extra/normalize.css:42]: sorry, the issue is bigger than just text inputs, it's just what is shown in the video. were you able to reproduce the issue on the text input?

File: src/extra/normalize.css [hunk 2]
```diff
@@ -46,13 +50,13 @@
   font-weight: var(--font-weight-9);
 }
 
-:where(h1) { 
-  font-size: var(--font-size-8); 
-  max-inline-size: var(--size-header-1); 
+:where(h1) {
+  font-size: var(--font-size-8);
+  max-inline-size: var(--size-header-1);
```
Comment (argyleink) [inline src/extra/normalize.css:55]: can you reduce the PR to only commits needed to fix the issue/feature please? 🙂

**Review by argyleink — CHANGES_REQUESTED** [review]
solution needs to be larger than just text inputs

File: src/extra/normalize.css [hunk 3]
```diff
@@ -37,6 +37,21 @@
...
 
+:where(a, button, input, textarea, summary) {
+  --outline-size: max(5px, 1px);
+  --outline-style: solid;
+  --outline-color: var(--blue-5);
+}
+
+:where(a, button, input, textarea, summary):focus-visible {
+  outline: var(--outline-size) var(--outline-style) var(--outline-color);
+  outline-offset: var(--outline-offset, var(--outline-size));
```
Comment (argyleink) [inline src/extra/normalize.css:48]: where is `--outline-offset` defined?

File: src/extra/normalize.css [hunk 4]
```diff
@@ -37,6 +37,21 @@
   outline-offset: 5px;
 }
 
+:where(a, button, input, textarea, summary) {
+  --outline-size: max(5px, 1px);
```
Comment (argyleink) [inline src/extra/normalize.css:41]: can you explain this a bit? how'd you come across it?

**Review by argyleink — COMMENTED** [review]
nice, ty for the revisions! 
i've added a couple questions and have done some light testing. 
i'm curious how you're testing the changes and verifying?

File: src/extra/normalize.css (hunk 4 above)
Comment (mrtuvn) [reply src/extra/normalize.css:41]: added some markup at docsite directory for testing from my mac

File: src/extra/normalize.css (hunk 3 above)
Comment (mrtuvn) [reply src/extra/normalize.css:48]: if --outline-offset missed or not define by user the second param will used instead

File: src/extra/normalize.css (hunk 4 above)
Comment (mrtuvn) [reply src/extra/normalize.css:41]: just add some local custom variables for easier to understand

File: src/extra/normalize.css (hunk 3 above)
Comment (mrtuvn) [reply src/extra/normalize.css:48]: In case redefine offset we can do like this
```css
input:focus-visible {
  --outilne-offset: 5px;
}
```

**Comment by argyleink** [comment]
`max(5px, 1px)` works because it sets the base value to the same as the focus value `5px`, really not transitioning any distance anymore and therefore not reproducing the bug. 

i think this is a `wont fix` in open props, instead i'll log the bug on webkit's bug repo to see if they'll fix it. like, if you hover over the poorly painted animated outlined element, the outline fixes itself. seems like a paint / animation bug that we can get fixed upstream for everyone instead of patching it with a custom style.

i appreciate you hackin on this and making the PR! ❤️  it forced me to investigate the issue further and establish a stronger understanding of the problem.

**Pull request closed without merging**